torchdynamo/optimizations/python_key.py

```diff
@@ -160,6 +157,7 @@ def unpack(x):
 
     with pythonkey_decompose({**aot_autograd_decompositions, **decompositions}):
         tracer: torch.fx.Tracer = PythonKeyTracer()
+        proxy_mode = ProxyTorchDispatchMode(tracer)
```

Comment:
We'll merge this as is. But can someone explain to me why we have this duplicate implementation of `make_fx` here? Can we dump this code and use most of `make_fx` to do this?

Reply:
I don't have history, but @Chillee and I were talking about just merging most of the stuff in here with the python key tracing in fx.

Reply:
yes plz

Reply:
More context is here https://github.com/pytorch/torchdynamo/issues/271

This version was originally created to capture factory functions, which at the time AOT Autograd didn't support.